include/service/entities/plugin_service.js

```diff
@@ -1079,7 +1139,13 @@ module.exports = function PluginServiceModule(pb) {
                 if(!cached_plugin || !cached_plugin.details) {
                     return PluginService.loadDetailsFile(PluginService.getDetailsPath(plugin.dirName), function (err, loadedDetails) {
                         details = loadedDetails;
-                        callback(err, !!details);
+                        if (autoSync && !pb.util.isError(err) && details) {
+                            var pluginService = new PluginService({site: site});
```

Comment:
The reality of the situation is that this function is WAY to big and should be moved to its own prototype at some point in the near future.  Until then i can live with it.  However, you are correct.  If you could just add an inline comment with a TODO to address this when that piece is broken out I would appreciate it.